simplq/src/store/queues/queuesAsyncActions.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable no-param-reassign */
+import useAuth, { makeAuthedRequest } from 'api/auth';
+
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { QueueRequestFactory } from 'api/requestFactory';
+import * as RequestFactory from 'api/requestFactory';
+
+/**
+ * A hook to access the fetchQuees async action creator.
+ *
+ * @returns — fetchQueues async async action creator
+ *
+ * @example
+ *
+ * import useFetchQueues from 'store/queues'
+ *
+ * const fetchQueues = useFetchQueues()
+ */
+const useFetchQueues = () => {
+  const auth = useAuth();
+
+  const fetchQueues = createAsyncThunk('queues/requestStatus', async () => {
+    if (!auth || !auth.isAuthenticated) {
+      return { queues: [] };
+    }
```

Comment:
Do we need `!auth` here? We should also add `auth. isLoading` here, I think.

Also leave this comment:
```
// TODO: Remove `!auth.isAuthenticated` after resolving https://github.com/SimplQ/simplQ-frontend/issues/477 
```

Reply:
We definitely do not need `!auth`, it is guaranteed to be there by the hook. There was a lot of copy-paste happening so there will be a lot to remove 😊 I haven't explored what exactly should happen when not authenticated, so that should be implemented later.